### PR TITLE
entrypoint cleanup + non-systemd-host fix

### PR DESCRIPTION
**Pull request by BenTheElder**
see https://github.com/kubernetes-sigs/kind/issues/2765, https://github.com/kubernetes-sigs/kind/pull/2743#discussion_r868931694

### CODE REVIEW

File: images/base/files/usr/local/bin/entrypoint [hunk 1]
```diff
@@ -55,43 +55,43 @@ validate_userns() {
 }
 
 overlayfs_preferrable() {
-	if [[ -z "$userns" ]]; then
```
Comment (BenTheElder) [inline images/base/files/usr/local/bin/entrypoint:58]: tabs => double space (like the rest of the file)

File: images/base/files/usr/local/bin/entrypoint [hunk 2]
```diff
@@ -225,9 +227,7 @@ fix_cgroup() {
   # See: https://d2iq.com/blog/running-kind-inside-a-kubernetes-cluster-for-continuous-integration
   # Capture initial state before modifying
   #
-  # Basically we're looking for the cgroup-path for the cpu controller for the
```
Comment (BenTheElder) [inline images/base/files/usr/local/bin/entrypoint:228]: things got re-arranged, and this comment got out of place.

File: images/base/files/usr/local/bin/entrypoint [hunk 3]
```diff
@@ -298,9 +298,15 @@ fix_cgroup() {
...
   echo "${cgroup_subsystems}" |
   while IFS= read -r subsystem; do
-    mount_kubelet_cgroup_root "/kubelet" "${subsystem}"
-    mount_kubelet_cgroup_root "/kubelet.slice" "${subsystem}"
+    mount_kubelet_cgroup_root /kubelet "${subsystem}"
+    mount_kubelet_cgroup_root /kubelet.slice "${subsystem}"
   done
+  # workaround for hosts not running systemd
+  # we only do this for kubelet.slice because it's not relevant when not using
+  # the systemd cgroup driver
```
Comment (BenTheElder) [inline images/base/files/usr/local/bin/entrypoint:306]: functional change here

File: images/base/files/usr/local/bin/entrypoint [hunk 4]
```diff
@@ -234,7 +234,7 @@ fix_cgroup() {
   # See: https://man7.org/linux/man-pages/man7/cgroups.7.html
   echo 'INFO: fix cgroup mounts for all subsystems'
   local cgroup_subsystems
-  cgroup_subsystems=$(findmnt -lun -o source,target -t cgroup | grep "${current_cgroup}" | awk '{print $2}')
+  cgroup_subsystems=$(findmnt -lun -o source,target -t cgroup | grep -F "${current_cgroup}" | awk '{print $2}')
```
Comment (BenTheElder) [inline images/base/files/usr/local/bin/entrypoint:237]: -F / --fixed-strings means match fixed strings, not patterns / regexes.
we're intending to match literal values

File: images/base/files/usr/local/bin/entrypoint (hunk 3 above)
Comment (aojea) [reply images/base/files/usr/local/bin/entrypoint:306]: @AkihiroSuda @kolyshkin can you please chime in here, I'm really in deep waters here and I think that a another pair of eyes will be nice for this change ... the cgroup glue keeps growing ...

Comment (BenTheElder) [reply images/base/files/usr/local/bin/entrypoint:306]: This doesn't work, wishful thinking :-)

https://github.com/kubernetes-sigs/kind/issues/2766#issuecomment-1126524575

I need to get a v1 host going again and take a peek at what exactly systemd is up to with the pseudo-subsystem.

**Comment by BenTheElder** [comment]
s390x flake https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kind/2767/pull-kind-build/1527143332810067968

**Comment by BenTheElder** [comment]
`bentheelder/kind-node:v1.24.0@sha256:cf92db251fb3f420634ec46d9c222812fee4d75f1e0105a325e57679b0f57370` is built with this change

I've tested it in colima xref https://github.com/kubernetes-sigs/kind/issues/2778, and this works now.

File: images/base/files/usr/local/bin/entrypoint (hunk 3 above)
Comment (BenTheElder) [reply images/base/files/usr/local/bin/entrypoint:306]: I setup colima on an old personal machine to debug, this _and_ a real slice works https://github.com/kubernetes-sigs/kind/pull/2767/commits/0ffcf8d6959f98ed038a925830b25f1f77480838.

**Comment by BenTheElder** [comment]
Confirmed WSL 2 https://github.com/kubernetes-sigs/kind/issues/2766#issuecomment-1131477890

I'll bump the node image and then we should ship this to v0.14

File: images/base/files/etc/systemd/system/kubelet.slice [hunk 5]
```diff
@@ -0,0 +1,7 @@
+[Unit]
+Description=slice used to run Kubernetes / Kubelet
+Before=slices.target
+
+[Slice]
+MemoryAccounting=true
```
Comment (aojea) [inline images/base/files/etc/systemd/system/kubelet.slice:6]: Are these parameters ***Accounting needed twice?

Comment (BenTheElder) [reply images/base/files/etc/systemd/system/kubelet.slice:6]: yes these parameters are for the slice. group vs specific unit.

https://www.freedesktop.org/software/systemd/man/systemd.slice.html

**Comment by BenTheElder** [comment]
Passed across github actions with the new node image.

Tested locally by me in colima and cgroups v2 ~debian. 
Also tested by others on rancher desktop and WSL 2

Base image already tested in all the pull-kubernetes* jobs across k8s versions in prow.

**Comment by aojea** [comment]
👌👍

**Comment by BenTheElder** [comment]
full test runs in #2780, not sure why 1.22 flaked but it only flaked here and not on the 2 subsequent PRs.

**Pull request merged**